beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/hive/HiveClient.java

```diff
@@ -0,0 +1,111 @@
+
+  private CloseableMetaStoreClient client;
+  private final boolean dryRunEnabled;
+
+  public HiveClient(CloseableMetaStoreClient client, boolean dryRunEnabled) {
+    this.client = client;
+    this.dryRunEnabled = dryRunEnabled;
+  }
+
+  /**
+   * Will delete the table metadata. Error is not thrown if table not found.
+   * 
+   * @param database
+   * @param tableName
+   */
+  public boolean deleteMetadata(String database, String tableName) {
+    boolean tableDeleted = true;
+    if (dryRunEnabled) {
+      log.info("Dry run - deleting metadata for table \"{}.{}\"", database, tableName);
+    } else {
+      log.info("Deleting metadata for table \"{}.{}\"", database, tableName);
+      try {
+        client.dropTable(database, tableName);
+      } catch (NoSuchObjectException e) {
+        log.info("Could not delete metadata. Table not found: \"{}.{}\"", database, tableName);
+        tableDeleted = false;
+      } catch (TException e) {
+        throw new BeekeeperException(
+            "Unexpected exception when deleting metadata: \"" + database + "." + tableName + "\".",
+            e);
+      }
+    }
+    return tableDeleted;
+  }
+
+  /**
+   * Will drop the partition from the table. Error is not thrown if table or partition are not found.
+   * 
+   * @param databaseName
+   * @param tableName
+   * @param partitionValue expected format: "event_date=2020-01-01/event_hour=0/event_type=A"
+   */
+  public boolean dropPartition(String databaseName, String tableName, String partitionName) {
+    boolean partitionDeleted = true;
+    if (dryRunEnabled) {
+      log.info("Dry run - dropping partition \"{}\" from table \"{}.{}\"", partitionName, databaseName, tableName);
+    } else {
+      log.info("Dropping partition \"{}\" from table \"{}.{}\"", partitionName, databaseName, tableName);
+      if (tableExists(databaseName, tableName)) {
+        try {
+          client.dropPartition(databaseName, tableName, partitionName, false);
+        } catch (NoSuchObjectException e) {
+          log
+              .info("Could not drop partition \"{}\" from table \"{}.{}\". Partition does not exist.", partitionName,
+                  databaseName, tableName);
+          partitionDeleted = false;
+        } catch (TException e) {
+          throw new BeekeeperException("Unexpected exception when dropping partition \""
+              + partitionName
+              + "\" from table: \""
+              + databaseName
+              + "."
+              + tableName
+              + "\".", e);
+        }
+      } else {
+        log.info("Could not drop partition from table \"{}.{}\". Table does not exist.", databaseName, tableName);
+        partitionDeleted = false;
+      }
+    }
+    return partitionDeleted;
+  }
+
+  private boolean tableExists(String databaseName, String tableName) {
+    try {
+      return client.tableExists(databaseName, tableName);
+    } catch (TException e) {
+      return false;
```

Comment:
Do you not want to rethrow this? Surely if this happens something has gone wrong and maybe it's not safe to assume that the table doesn't exist?